netaddr.go

```diff
@@ -411,10 +385,10 @@ func (ip IP) IsLinkLocalUnicast() bool {
 // it will return false.
 func (ip IP) IsLoopback() bool {
 	if ip.Is4() {
-		return ip.a[o4+0] == 127
+		return uint8(ip.lo>>24) == 127
```

Comment:
I'd like to avoid a number of these magic constants like "24".

How about some accessors to get at the bytes of an IPv4 address? I think they should inline away to nothing. Either one but really four might be more readable at call sites:

```go
func (ip *IP) v4a() uint8 { return uint8(ip.lo >> 24) }
func (ip *IP) v4b() uint8 { return uint8(ip.lo >> 16) }
func (ip *IP) v4c() uint8 { return uint8(ip.lo >> 8) }
func (ip *IP) v4d() uint8 { return uint8(ip.lo) }
```

(That's assuming the pointer receiver inlines better; it might not matter in which case value receiver is more proper)

Reply:
My personal preference is for `v4(n uint8)` and `v6(n uint8)`, for consistency. I changed the various sites to use those, except for `Is4in6` where we're doing a 48-bit comparison, and it feels silly to break that out into 6 bytewise compares.

Per above, unfortunately I can't easily inspect what the compiler produces, because GOSSAFUNC triggers a nil dereference panic when I try to analyze any function that includes calls to v4/v6.